app/screens/settings/notification_mention/mention_settings.tsx

```diff
@@ -131,30 +109,50 @@ const MentionSettings = ({componentId, currentUser, isCRTEnabled}: MentionSectio
 
         close();
     }, [
-        canSaveSettings,
         channelMentionOn,
         firstNameMentionOn,
+        usernameMentionOn,
         mentionKeywords,
         notifyProps,
+        mentionProps,
         replyNotificationType,
         serverUrl,
+        currentUser,
     ]);
 
-    const onToggleFirstName = useCallback(() => {
-        setFirstNameMentionOn((prev) => !prev);
-    }, []);
+    function handleFirstNameToggle() {
+        setFirstNameMentionOn(!firstNameMentionOn);
+    }
 
-    const onToggleUserName = useCallback(() => {
-        setUsernameMentionOn((prev) => !prev);
-    }, []);
+    function handleUsernameToggle() {
+        setUsernameMentionOn(!usernameMentionOn);
+    }
+
+    function handleChannelToggle() {
+        setChannelMentionOn(!channelMentionOn);
+    }
+
+    function handleMentionKeywordsInputChanged(text: string) {
+        // Temp fix remove later >>>>>>>>>>>
+        if (text[text.length - 1] === ',') {
+            setMentionKeywordsInput('');
+        } else {
+            setMentionKeywordsInput(text);
+        }
+    }
 
-    const onToggleChannel = useCallback(() => {
-        setChannelMentionOn((prev) => !prev);
-    }, []);
+    function handleMentionKeywordEntered() {
+        // Check if the keyword is not empty and not already in the list
+        if (mentionKeywordsInput.length > 0 && !mentionKeywords.includes(mentionKeywordsInput)) {
+            const formattedKeyword = mentionKeywordsInput.trim().replace(/ /g, '');
```

Comment:
Why removing all spaces? And if doing so, shouldn't be done before checking the length?

Reply:
yes correct changed